spec/lib/oauth/pre_authorization_spec.rb

```diff
@@ -331,6 +331,22 @@
         expect(pre_auth).not_to be_authorizable
       end
 
+      context "when pkce_code_challenge_methods is set to none" do
+        before do
+          Doorkeeper.configure do
+            pkce_code_challenge_methods []
+          end
+        end
+
+        it "rejects plain as a code_challenge_method" do
+          attributes[:code_challenge] = "a45a9fea-0676-477e-95b1-a40f72ac3cfb"
+          attributes[:code_challenge_method] = "plain"
+
+          expect(pre_auth).to_not be_authorizable
+          expect(pre_auth.error_response.description).to eq("There are no acceptable code_challenge methods")
```

Comment:
Also, we should use `code_challenge` or `code challenge` consistently.

Reply:
Good catch, this comment made me realize that we should probably be calling it `code_challenge_method` since that is the actual param, so I made that change in [d244e69](https://github.com/doorkeeper-gem/doorkeeper/pull/1747/commits/d244e6937dbe3f0e6194ae15438d6cbafd30092d)